### DISCUSSION TITLE
Wrong fs type, bad option, bad superblock on /dev/longhorn/xxx, missing codepage or helper program, or other error.

### DISCUSSION THREAD

**Discussion by MagicFollower** [longhorn/longhorn#4568] [General]
【Longhorn v1.3.1】
When I restore pv/pvc from Backup data, I mount the restored pv into a new pod, I got this error.
What's causing this problem? My backup (NFS) host filesystem is XFS and all node filesystems are XFS.

I found a previous similar question, but there was no solution to that question.（https://github.com/longhorn/longhorn/discussions/3472）
👇👇👇
![image](https://user-images.githubusercontent.com/89330933/190218593-b4cb0ec8-83f3-4ece-b09e-514ef2c46b70.png)

**Comment by derekbit** [comment]
@MagicFollower 
Can you provide the information how to reproduce it?
BTW, I would suggest you using ext4 rather than zfs.

**Comment by derekbit** [comment]
I feel this might be related to the filesystem corruption.
The backup job used a corrupted replica and then failed to restore it.
ref: https://github.com/longhorn/longhorn/issues/3597

**Reply by MagicFollower** [reply]
Thank you:)  I have solved this problem, but there is one more question I want to ask. For the experimental environment, I always operate longhron in RancherUI. During initialization, I found that longhron will use the ext4 file system for the created pv by default, and all backup records will exist in this file system type. When restoring from backup data, if you choose to accidentally select xfs, the backup data will not be restored. After I restore the backup file, change the file system type to ext4 and it can be restored normally.

The problem now is that the file system type of my local node is xfs, the pv block file system type created by longhorn by default is ext4, and the pv block of ext4 type is created on the xfs file system, will there be any potential problems? Or do I need to change the filesystem type where longhron creates pv blocks by default to xfs (the same filesystem type as my host).

**Reply by derekbit** [reply]
> if you choose to accidentally select xfs, the backup data will not be restored. After I restore the backup file, change the file system type to ext4 and it can be restored normally.

I see. We might be able to improve this (e.g. error message or select a correct fs automatically) to avoid the error.

> the pv block of ext4 type is created on the xfs file system, will there be any potential problems

No. longhorn filesystem is not releated to underlying xfs filesystem.

**Reply by MagicFollower** [reply]
🚀🚀🚀👍

**Reply by egoisz** [reply]
Can you tell me how to solve this problem?
I see the ref: https://github.com/longhorn/longhorn/issues/3597
Does I need to "mkfs.xfs -b size=1024 /dev/sda2 -f" like this to change the block size?
